biggus/__init__.py

```diff
@@ -2043,33 +2042,51 @@ def ndarrays(arrays):
     return engine.ndarrays(*arrays)
 
 
-#: The maximum number of bytes to allow when processing an array in
-#: "bite-size" chunks. The value has been empirically determined to
+#: The maximum number of bytes per chunk to allow when processing an array in
+#: "bite-size" chunks. Chunks are determined by the _all_slices function, where
+#: an assumption is made that data type nbytes is 8.
+#: The value has been empirically determined to
 #: provide vaguely near optimal performance under certain conditions.
-MAX_CHUNK_SIZE = 8 * 1024 * 1024
+MAX_CHUNK_SIZE = 8 * 1024 * 1024 * 2
 
 
 def _all_slices(array):
-    return _all_slices_inner(array.dtype.itemsize, array.shape)
+    return _all_slices_inner(array.shape)
 
 
-def _all_slices_inner(item_size, shape, always_slices=False):
+def _all_slices_inner(shape, always_slices=False):
     # Return the slices for each dimension which ensure complete
     # coverage by chunks no larger than MAX_CHUNK_SIZE.
     # e.g. For a float32 array of shape (100, 768, 1024) the slices are:
     #   (0, 1, 2, ..., 99),
     #   (slice(0, 256), slice(256, 512), slice(512, 768)),
     #   (slice(None)
-    nbytes = item_size
+
+    # Fix the item size to 8 bytes, as this is part of the definition of
+    # MAX_CHUNK_SIZE. nbytes will be updated as we traverse the dimensions of
+    # shape so that it equals the number of bytes that one item in the current
+    # dimension represents.
+    nbytes = 8
```

Comment:
It doesn't make sense to me to keep the concept of bytes. Really you're moving from counting bytes to counting array elements.